nitro-protocol/contracts/ForceMove.sol

```diff
@@ -299,24 +180,16 @@ contract ForceMove is IForceMove, StatusManager {
     }
 
     /**
-     * @notice Validates input for several external methods.
-     * @dev Validates input for several external methods.
-     * @param numParticipants Length of the participants array
-     * @param numStates Number of states submitted
-     * @param numSigs Number of signatures submitted
-     * @param numWhoSignedWhats whoSignedWhat.length
+     * @notice Validates related to participants and states input for several external methods.
+     * @dev Validates related to participants and states input for several external methods.
+     * @param numParticipants Length of the participants array.
+     * @param numStates Number of states submitted.
      */
     function requireValidInput(
         uint256 numParticipants,
-        uint256 numStates,
-        uint256 numSigs,
-        uint256 numWhoSignedWhats
+        uint256 numStates
```

Comment:
We should think about whether this check is necessary or not. 

The old check was important, but I think we don't need this check anymore since we are making the protocol much more open with respect to what can constitute a support proof. In particular, we defer to the application rules to decide. It might be that most (or even all) applications would have the same check inside. But it probably doesn't belong here. 

There might also be an interesting application that has more states in the support proof than there are participants in the channel. I can't think of an example. But I also can't think why this could be ruled out -- can you @nksazonov or @andrewgordstewart ?

Reply:
Agree with @andrewgordstewart . This check belongs to deciding what constitutes a support proof, which we give away to DApp.